internal/pkg/service/stream/storage/level/local/writer/base.go

```diff
@@ -1,105 +0,0 @@
-package writer
-
-import (
-	"context"
-
-	"github.com/benbjohnson/clock"
-
-	"github.com/keboola/keboola-as-code/internal/pkg/log"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/mapping/table/column"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/compression"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/volume/disksync"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/local/writer/writechain"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
-)
-
-type chain = writechain.Chain
-
-type syncer = disksync.Syncer
-
-// BaseWriter implements common logic for all file types.
-type BaseWriter struct {
-	*chain
-	*syncer
-	events   *Events
-	logger   log.Logger
-	slice    *model.Slice
```

Comment:
Originally, there was `BaseWriter` and `CSVWriter` was using it.

Now, there is `Writer`, and its dependency is `FormatWriterFactory`.

The `Writer` creates on initialization the `FormatWriter` (CSV writer) and the uses it.